From a51bea5534076916587c4bebc2299d7881d6fad7 Mon Sep 17 00:00:00 2001
From: Max Ihlenfeldt <max@igalia.com>
Date: Mon, 5 Aug 2024 09:33:42 +0000
Subject: [PATCH] Disable crabbyavif to fix build errors

It seems the way upstream sets up bindgen does not play well with
cross-compiling in certain configurations. Currently, this only affects
building the crabbyavif third-party library, which is still being tested
behind a feature flag. Until we find a way to fix the bindgen setup
(either upstream or downstream), we just disable crabbyavif by reverting
the CL that added it to the build, https://crrev.com/c/5542730.

Upstream-Status: Inappropriate [upstream ticket https://crbug.com/357017325]
Signed-off-by: Max Ihlenfeldt <max@igalia.com>
---
 .../renderer/platform/image-decoders/BUILD.gn  |  8 +-------
 .../platform/image-decoders/image_decoder.cc   | 18 ++++--------------
 2 files changed, 5 insertions(+), 21 deletions(-)

diff --git a/third_party/blink/renderer/platform/image-decoders/BUILD.gn b/third_party/blink/renderer/platform/image-decoders/BUILD.gn
index 6dbddaa2e3..2b5f660196 100644
--- a/third_party/blink/renderer/platform/image-decoders/BUILD.gn
+++ b/third_party/blink/renderer/platform/image-decoders/BUILD.gn
@@ -82,12 +82,9 @@ component("image_decoders") {
     sources += [
       "avif/avif_image_decoder.cc",
       "avif/avif_image_decoder.h",
-      "avif/crabbyavif_image_decoder.cc",
-      "avif/crabbyavif_image_decoder.h",
     ]

     deps += [
-      "//third_party/crabbyavif",
       "//third_party/libavif",
     ]
   }
@@ -136,9 +133,6 @@ source_set("unit_tests") {
   }

   if (enable_av1_decoder) {
-    sources += [
-      "avif/avif_image_decoder_test.cc",
-      "avif/crabbyavif_image_decoder_test.cc",
-    ]
+    sources += [ "avif/avif_image_decoder_test.cc" ]
   }
 }
diff --git a/third_party/blink/renderer/platform/image-decoders/image_decoder.cc b/third_party/blink/renderer/platform/image-decoders/image_decoder.cc
index bdbbe08142..8b08f320cd 100644
--- a/third_party/blink/renderer/platform/image-decoders/image_decoder.cc
+++ b/third_party/blink/renderer/platform/image-decoders/image_decoder.cc
@@ -37,7 +37,6 @@
 #include "media/media_buildflags.h"
 #include "skia/ext/cicp.h"
 #include "third_party/blink/public/common/buildflags.h"
-#include "third_party/blink/public/common/features.h"
 #include "third_party/blink/public/platform/platform.h"
 #include "third_party/blink/renderer/platform/image-decoders/bmp/bmp_image_decoder.h"
 #include "third_party/blink/renderer/platform/image-decoders/fast_shared_buffer_reader.h"
@@ -53,7 +52,6 @@

 #if BUILDFLAG(ENABLE_AV1_DECODER)
 #include "third_party/blink/renderer/platform/image-decoders/avif/avif_image_decoder.h"
-#include "third_party/blink/renderer/platform/image-decoders/avif/crabbyavif_image_decoder.h"
 #endif

 namespace blink {
@@ -198,9 +196,7 @@ String SniffMimeTypeInternal(scoped_refptr<SegmentReader> reader) {
     return "image/bmp";
   }
 #if BUILDFLAG(ENABLE_AV1_DECODER)
-  if (base::FeatureList::IsEnabled(blink::features::kCrabbyAvif)
-          ? CrabbyAVIFImageDecoder::MatchesAVIFSignature(fast_reader)
-          : AVIFImageDecoder::MatchesAVIFSignature(fast_reader)) {
+  if (AVIFImageDecoder::MatchesAVIFSignature(fast_reader)) {
     return "image/avif";
   }
 #endif
@@ -310,15 +306,9 @@ std::unique_ptr<ImageDecoder> ImageDecoder::CreateByMimeType(
                                                 max_decoded_bytes);
 #if BUILDFLAG(ENABLE_AV1_DECODER)
   } else if (mime_type == "image/avif") {
-    if (base::FeatureList::IsEnabled(blink::features::kCrabbyAvif)) {
-      decoder = std::make_unique<CrabbyAVIFImageDecoder>(
-          alpha_option, high_bit_depth_decoding_option, color_behavior,
-          aux_image, max_decoded_bytes, animation_option);
-    } else {
-      decoder = std::make_unique<AVIFImageDecoder>(
-          alpha_option, high_bit_depth_decoding_option, color_behavior,
-          aux_image, max_decoded_bytes, animation_option);
-    }
+    decoder = std::make_unique<AVIFImageDecoder>(
+        alpha_option, high_bit_depth_decoding_option, color_behavior,
+        aux_image, max_decoded_bytes, animation_option);
 #endif
   }

